Data processing

hillaryj opened the pull request:
Adds data munging script for all source data files

Commit:
Initial commit

Commit:
Updates for several file types

Commit:
Adds group volume files

Processes group volume files

Commit:
Processes all file types

All current file types now implemented.

marteki commented:
Could we format the borrower risk profile categories the same way that they will be in the page content and in the charts? Some of them don't have the correct spacing or en dashes.

* Deep subprime
* Subprime
* Near-prime
* Prime
* Super-prime

marteki commented:
Also, could we use the same categories for age groups that we're using in the content and in the charts?

* Younger than 30
* Age 30-44
* Age 45-64
* Age 65 and older

willbarton commented on process_incoming_data.py:
```diff
@@ -0,0 +1,604 @@
+        if market is None:
+            print("Found file '{}' does not specify market".format(filename))
+            failures.append(filename)
+            continue
```
Is it important to know the failures in real-time? If not, it might be nice to record the file and the reason for failure as tuples in the `failures` list so that you can print them in a single place (L233) below. But that would just be my preference. It's fine like it is too.

willbarton commented on process_incoming_data.py:
```diff
@@ -0,0 +1,604 @@
+            if cond:
+                successes.append(filename)
+            else:
+                failures.append(filename)
```
Re: the note above, that would let you add the stage that failure occurred here too.

willbarton commented on process_incoming_data.py:
```diff
@@ -0,0 +1,604 @@
+    # Check if any data exists besides column headers
+    if len(data) > 1:
+        return True, data
+    else:
```
If you're returning in the `if` above, the `else` is not needed.

willbarton commented on process_incoming_data.py:
```diff
@@ -0,0 +1,604 @@
+    # Check if any data exists besides column headers
+    if len(data) > 1:
+        return True, data
+    else:
```
Same as above, with the `else`.

willbarton commented on process_incoming_data.py:
```diff
@@ -0,0 +1,604 @@
+    # Check if any data exists besides column headers
+    if len(data) > 1:
+        return True, data
+    else:
```
Same.

willbarton commented on process_incoming_data.py:
```diff
@@ -0,0 +1,604 @@
+    # Check if any data exists besides column headers
+    if len(data) > 1:
+        return True, data
+    else:
```
💣

willbarton reviewed (commented):
This looks so far as I can tell! Just a few very, very minor things.

I will make a gentle request for tests. They'd help illustrate what's meant to be going on here, and make it easier in the future to determine why something fails if the data changes in subtle ways. That, and there are the testing goats:

![d9ae1ea36978f911cfcee26ddf1a8864](https://cloud.githubusercontent.com/assets/10562538/22427989/4c6e1f56-e6d3-11e6-8366-ac34872f46af.jpg)

However, I don't have any reason this shouldn't merge from just reviewing the code. 👍

Commit:
Corrects text labels and improves formatting

Updates output text in group names to follow agency guidelines. Also
incorporates comments on some code formatting as well as consolidates
two sections of helper functions into one.

Commit:
Removed some extraneous commented-out code

Commit:
Removes snyk so Travis stops failing

Commit:
Removed text fixes that didn't change anything

Some text fixes changed nothing, so those have been removed to reduce
processing load.

Commit:
Revert "Removes snyk so Travis stops failing"

This reverts commit 1b4eb1af96c0b609b27ff117c2fc0f479cc1edfa.

Commit:
Removes snyk to stop failing
